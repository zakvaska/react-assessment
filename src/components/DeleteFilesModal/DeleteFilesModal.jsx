import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import Modal from '../Modal/Modal';
import {
  ModalBody,
  ModalHeader,
  ModalFooter,
  ButtonList,
  StyledButton
} from '../styled-components/Task';

const DeleteFilesModal = ({ modalCloseHandler, onConfirm, onCancel }) => (
  <Modal modalCloseHandler={modalCloseHandler}>
    <ModalHeader>
      <FontAwesomeIcon icon={faExclamationTriangle} color="#d0011b" />
      <div>Are you sure you want to delete all of your files?</div>
    </ModalHeader>
    <ModalBody>This action can not be undone.</ModalBody>
    <ModalFooter>
      <ButtonList justify="flex-end" marginRight="1rem">
        <StyledButton type="button" onClick={onConfirm}>Yes</StyledButton>
        <StyledButton type="button" onClick={onCancel}>No</StyledButton>
      </ButtonList>
    </ModalFooter>
  </Modal>
);

DeleteFilesModal.propTypes = {
  modalCloseHandler: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired
};

export default DeleteFilesModal;
