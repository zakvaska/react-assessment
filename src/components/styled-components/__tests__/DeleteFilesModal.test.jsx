import React from 'react';
import { render } from '@testing-library/react';
import DeleteFilesModal from '../../DeleteFilesModal/DeleteFilesModal';
import 'jest-styled-components';

it('DeleteFilesModal renders correctly', () => {
  const handleClose = jest.fn();
  const handleConfirm = jest.fn();
  const handleCancel = jest.fn();
  const { container } = render(
    <DeleteFilesModal 
      modalCloseHandler={handleClose}
      onConfirm={handleConfirm}
      onCancel={handleCancel}
    />
  );
  expect(container).toMatchSnapshot();
});
