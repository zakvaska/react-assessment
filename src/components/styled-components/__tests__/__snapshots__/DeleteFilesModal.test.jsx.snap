// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFilesModal renders correctly 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > button:not(:last-of-type) {
  margin-right: 1rem;
}

.c5 > button {
  min-width: 7rem;
}

.c6 {
  color: #24619a;
  background: #e8f4ff;
  border: 1px solid transparent;
  padding: 0.7rem 1rem;
  border-radius: 0.3rem;
  margin-top: 2rem;
  cursor: pointer;
  font-family: sans-serif;
  font-size: 1rem;
  font-weight: bold;
  -webkit-transition: background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
}

.c6:hover:not(:disabled) {
  background: white;
  border: 1px solid #24619a;
}

.c6:disabled {
  color: #24619a73;
  cursor: default;
}

.c0 {
  position: absolute;
  width: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  background-color: white;
  border: 1px solid #d10720;
  width: 35rem;
  border-radius: .75rem;
  margin: 10rem auto;
  padding: 1.5rem;
  color: black;
  text-align: start;
  font-family: sans-serif;
  font-size: 1.1rem;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #e1e1e1;
}

.c2 > div {
  padding-left: .5rem;
  font-weight: bold;
  color: black;
  font-family: sans-serif;
}

.c3 {
  font-family: sans-serif;
  font-size: 1rem;
  padding-top: 1rem;
}

.c4 {
  padding-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
          color="#d0011b"
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
        <div>
          Are you sure you want to delete all of your files?
        </div>
      </div>
      <div
        class="c3"
      >
        This action can not be undone.
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <button
            class="c6"
            type="button"
          >
            Yes
          </button>
          <button
            class="c6"
            type="button"
          >
            No
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
.c0 {
  max-height: 300px;
  overflow: scroll;
  margin-top: 15px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
